### ISSUE TITLE
Re-think how we cross-validate

### ISSUE THREAD

**Issue by rodluger** [rodluger/flatironhack#6]
Currently doing a grid search in each PLD order as a shortcut. Should optimize the full 3D function.

**Comment by rodluger** [comment]
The derivative of the RMS in the validation set with respect to Lambda is analytic. This can be used in a gradient descent / nonlinear optimizer in 3D. We may still have issues with the non-smoothness of the function, but this could save us a lot of time.

![img_20171025_110656](https://user-images.githubusercontent.com/9323819/32006450-d0ecf274-b974-11e7-80e7-a70dd650fc69.jpg)

**Comment by rodluger** [comment]
Currently thinking of optimizing full likelihood instead of CV.
